pallets/pool-registry/src/benchmarking.rs

```diff
@@ -11,140 +11,200 @@
 // MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 // GNU General Public License for more details.
 
-//! Module provides benchmarking for the Pools Pallet
-use frame_benchmarking::{account, benchmarks, impl_benchmark_test_suite};
+//! Module provides benchmarking for Loan Pallet
+use cfg_primitives::PoolEpochId;
+use cfg_traits::{InvestmentAccountant, InvestmentProperties, TrancheCurrency as _};
+use cfg_types::tokens::{CurrencyId, TrancheCurrency};
+use frame_benchmarking::{benchmarks, impl_benchmark_test_suite};
+use frame_support::traits::fungibles::Inspect;
 use frame_system::RawOrigin;
+use orml_traits::Change;
+#[cfg(feature = "runtime-benchmarks")]
+use pallet_pool_system::benchmarking::{
+	assert_input_tranches_match, assert_update_tranches_match, build_bench_input_tranches,
+	create_admin, create_investor, create_pool, get_pool, get_scheduled_update, get_tranche_id,
+	prepare_asset_registry, update_pool,
+};
+use pallet_pool_system::{
+	pool_types::PoolChanges,
+	tranches::{TrancheIndex, TrancheInput, TrancheMetadata, TrancheType, TrancheUpdate},
+};
+use sp_runtime::{
+	traits::{One, Zero},
+	Perquintill,
+};
 use sp_std::vec;
 
 use super::*;
 
+const CURRENCY: u128 = 1_000_000_000_000_000;
+const MAX_RESERVE: u128 = 10_000 * CURRENCY;
+
+const SECS_PER_HOUR: u64 = 60 * 60;
+const SECS_PER_DAY: u64 = 24 * SECS_PER_HOUR;
+const SECS_PER_YEAR: u64 = 365 * SECS_PER_DAY;
+
+const TRANCHE: TrancheIndex = 0;
+
 const POOL: u64 = 0;
 
 benchmarks! {
 	where_clause {
 	where
-		T: Config<PoolId = u64>,
+		T: Config<PoolId = u64,
+			  TrancheId = [u8; 16],
+			  Balance = u128,
+			  CurrencyId = CurrencyId> + pallet_investments::Config<InvestmentId = TrancheCurrency, Amount = u128>,
+		T: pallet_pool_system::Config<PoolId = u64,
+			  TrancheId = [u8; 16],
+			  Balance = u128,
+			  CurrencyId = CurrencyId,
+			  EpochId = PoolEpochId,
+			  Rate = <T as Config>::InterestRate,
+			  MaxTokenNameLength = <T as Config>::MaxTokenNameLength,
+			  MaxTokenSymbolLength = <T as Config>::MaxTokenSymbolLength,
+			  MaxTranches = <T as Config>::MaxTranches>,
+		<T as pallet_investments::Config>::Tokens: Inspect<T::AccountId, AssetId = CurrencyId, Balance = u128>,
```

Comment:
Don't wanna block this, but this binding basically forces us to use the investments pallet always in the registry.

Instead bindings should be based on traits of functionality rather than traits of pallets wherever possible.

Reply:
Hm, this part is just taken from the `pallet-pool-system` benchmarking without changing it (https://github.com/centrifuge/centrifuge-chain/blob/main/pallets/pool-system/src/benchmarking.rs#L47). 

I assume we would need to create a trait for it first to abstract away the investments pallet?